shared/dataflow/codeql/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -652,46 +773,65 @@ module MakeImplCommon<LocationSig Location, InputSig<Location> Lang> {
       module NoLocalCallContext {
         class LocalCc = Unit;
 
-        bindingset[c, cc]
-        LocalCc getLocalCc(DataFlowCallable c, CallContext cc) { any() }
+        bindingset[cc]
+        LocalCc getLocalCc(CallContext cc) { any() }
 
         bindingset[call, c]
         CallContextCall getCallContextCall(DataFlowCall call, DataFlowCallable c) {
           if recordDataFlowCallSiteDispatch(call, c)
-          then result = TSpecificCall(call)
+          then result = Input2::getSpecificCallContextCall(call, c)
           else result = TSomeCall()
         }
       }
 
       module LocalCallContext {
         class LocalCc = LocalCallContext;
 
-        bindingset[c, cc]
-        LocalCc getLocalCc(DataFlowCallable c, CallContext cc) {
-          result = getLocalCallContext(pragma[only_bind_into](pragma[only_bind_out](cc)), c)
+        private UnreachableSet getUnreachable(CallContext ctx) {
+          exists(UnreachableSetOption unreachable | ctx = TSpecificCall(_, _, unreachable) |
+            result = unreachable.asSome()
+          )
```

Comment:
nit; I would prefer
```ql
result = any(UnreachableSetOption unreachable | ctx = TSpecificCall(_, _, unreachable)).asSome()
```

Reply:
I slightly disagree, actually. I think the `exists` formulation is more readable, as the `.asSome()` is more prominent.